writers/local/local.go

```diff
@@ -72,69 +77,78 @@ func (l *Local) Setup(stream protocol.Stream, options *protocol.Options) error {
 
-	// Write some content to the file (optional)
-	if _, err := tempFile.Write([]byte("Hello, this is a temporary file!")); err != nil {
-		return err
-	}
+	// Setup S3 client if S3 configuration is provided
+	if l.config.Bucket != "" {
+		s3Config := aws.Config{
+			Region: aws.String(l.config.Region),
+		}
+		if l.config.AccessKey != "" && l.config.SecretKey != "" {
+			s3Config.Credentials = credentials.NewStaticCredentials(l.config.AccessKey, l.config.SecretKey, "")
+		}
+		sess, err := session.NewSession(&s3Config)
+		if err != nil {
+			return fmt.Errorf("failed to create AWS session: %w", err)
+		}
+		l.s3Client = s3.New(sess)
 
-	// Close the file
-	if err := tempFile.Close(); err != nil {
-		return err
+		basePath := filepath.Join(stream.Namespace(), stream.Name())
+		if l.config.Prefix != "" {
+			basePath = filepath.Join(l.config.Prefix, basePath)
+		}
+		l.s3KeyPath = filepath.Join(basePath, l.fileName)
 	}
 
-	// Delete the temporary file
-	return os.Remove(tempFile.Name())
+	return nil
 }
 
-func (l *Local) Write(ctx context.Context, record types.Record) error {
-	err := l.writer.FlushRowGroupWithContext(ctx)
-	if err != nil {
-		return err
-	}
-	err = func() error {
-		l.pqSchemaMutex.Lock()
-		defer l.pqSchemaMutex.Unlock()
-		return l.writer.AddData(record)
-	}()
-	if err != nil {
-		return fmt.Errorf("parquet write error: %s", err)
+// Write writes a record to the Parquet file.
+func (l *Local) Write(_ context.Context, record types.Record) error {
+	// Lock for thread safety and write the record
+	l.pqSchemaMutex.Lock()
+	defer l.pqSchemaMutex.Unlock()
+
+	if err := l.writer.AddData(record); err != nil {
+		return fmt.Errorf("parquet write error: %w", err)
 	}
 
 	l.records.Add(1)
 	return nil
 }
 
+// ReInitiationOnTypeChange always returns true to reinitialize on type change.
 func (l *Local) ReInitiationOnTypeChange() bool {
 	return true
 }
 
+// ReInitiationOnNewColumns always returns true to reinitialize on new columns.
 func (l *Local) ReInitiationOnNewColumns() bool {
 	return true
 }
 
-func (l *Local) EvolveSchema(_ map[string]*types.Property) error {
-	l.pqSchemaMutex.Lock()
-	defer l.pqSchemaMutex.Unlock()
+// Check validates local paths and S3 credentials if applicable.
+func (l *Local) Check() error {
+	// Validate the local path
+	if err := os.MkdirAll(l.config.Path, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create path: %w", err)
+	}
+
+	// Test directory writability
+	tempFile, err := os.CreateTemp(l.config.Path, "temporary-*.txt")
+	if err != nil {
+		return fmt.Errorf("directory is not writable: %w", err)
+	}
+	tempFile.Close()
+	os.Remove(tempFile.Name())
 
-	return l.writer.SetSchemaDefinition(l.stream.Schema().ToParquet())
+	// Validate S3 credentials if S3 is configured
+	if l.config.Bucket != "" && l.s3Client != nil {
+		if _, err := l.s3Client.ListBuckets(&s3.ListBucketsInput{}); err != nil {
+			return fmt.Errorf("failed to validate S3 credentials: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to validate S3 credentials: %s", err)
```

use notation %s in errors